modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/SubscribeManager.kt

```diff
@@ -225,15 +225,21 @@ class SubscribeManager @Inject constructor(
             .subscribeOn(Schedulers.io())
             .doOnSuccess { Timber.i("Downloaded colors success podcast $podcastUuid") }
             .onErrorReturn { Optional.empty() }
+        // keep expanded or collapsed header state
+        val isHeaderExpandedObservable = podcastDao.findByUuidRxMaybe(podcastUuid)
+            .subscribeOn(Schedulers.io())
+            .map { it.isHeaderExpanded }
+            .toSingle(true)
```

Comment:
I could be wrong, but it seems like this logic should be in [SubscribeManager.kt#L191](https://github.com/Automattic/pocket-casts-android/blob/main/modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/SubscribeManager.kt#L191). This method involves downloading the podcasts, and the other code seems to be related to merging with the existing podcast logic.